src/transformers/models/bert/modeling_flax_bert.py

```diff
@@ -890,63 +909,115 @@ def __call__(
         if head_mask is None:
             head_mask = jnp.ones((self.config.num_hidden_layers, self.config.num_attention_heads))
 
-        # Handle any PRNG if needed
-        rngs = {}
-        if dropout_rng is not None:
-            rngs["dropout"] = dropout_rng
+        if self.scope is None:
+            if not self._do_init and params is None:
+                raise ValueError(
+                    f"{self.__class__.__name__} was initialized with `_do_init=False` but `params` was provided."
+                )
 
-        inputs = {"params": params or self.params}
+            # Handle any PRNG if needed
+            rngs = {}
+
+            if dropout_rng is not None:
+                rngs["dropout"] = dropout_rng
+
+            params = params or self.params
+
+            if "_module" in params:
```

Comment:
Would it be more user friendly to allow params to be passed as a dict without the extra `"_module"` nesting, e.g.
```diff
- {"params": {"_module": params}
+ {"params": params}
```